src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ProtocolExtensions.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using OmniSharpFormattingOptions = OmniSharp.Extensions.LanguageServer.Protocol.Models.FormattingOptions;
+using OmniSharpPosition = OmniSharp.Extensions.LanguageServer.Protocol.Models.Position;
+using OmniSharpRange = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+using OmniSharpTextEdit = OmniSharp.Extensions.LanguageServer.Protocol.Models.TextEdit;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    internal static class ProtocolExtensions
+    {
+        public static Position ToLSPPosition(this OmniSharpPosition position)
```

Comment:
> when we eventually move out of process we don't want our VS pieces to be dependent on the O# binaries.

But don't we already depend on our LanguageServer project which is how O# dependency is brought in. Does this mean we should also remove the LanguageServer dependency?

Reply:
> Because O# `FormattingOptions` is itself a dictionary where as the VS LSP `FormattingOptions` contains a property called `OtherOptions` which is a dictionary.

Interesting, is that an LSP spec violation on VS' part or O#'s part?



> But don't we already depend on our LanguageServer project which is how O# dependency is brought in.

Yup, we depend on it to create the language server. What I'm trying to say is I want to limit the API surface we use in VS from O# so when we later try to move out of process we don't bind ourselves to VS other than the initial creation of the VS language server.



> Does this mean we should also remove the LanguageServer dependency?

Nope, just means that our public API surface area in the language server library should only contain the bare minimum.

Reply:
It appears the violation is on VS' part,
![image](https://user-images.githubusercontent.com/1579269/78400284-3d81de00-75ab-11ea-9292-48ffb811f800.png)